dali/pipeline/operators/decoder/nvjpeg_decoder_decoupled_api.h

```diff
@@ -204,11 +211,19 @@ class nvJPEGDecoder : public Operator<MixedBackend> {
       const auto &in = ws->Input<CPUBackend>(0, i);
       const auto file_name = in.GetSourceInfo();
       cudaStream_t stream = ws->stream();
-      thread_pool_.DoWorkWithID(std::bind(
-            [this, i, file_name, stream, &in, &output](int idx, int tid) {
-              SampleWorker(i, file_name, in.size(), tid,
-                              in.data<uint8_t>(), output.mutable_tensor<uint8_t>(i));
-            }, i, std::placeholders::_1));
+      auto *output_data = output.mutable_tensor<uint8_t>(i);
+      auto dims = output_shape_[i];
+      ImageCache::ImageShape shape = {dims[0], dims[1], dims[2]};
```

Comment:
Might be interesting to have an iterator range ctor for `kernels::TensorShape`

Reply:
I agree but probably not part of this PR